### ISSUE TITLE
[TypeChecker] Bad cast of argument tuple in abi.encodeCall

### ISSUE THREAD

**Issue by bshastry** [ethereum/solidity#13204]
```
contract C{
  function f() { 
    abi.encodeCall(this.f, !());
  }
}
```

throws

https://github.com/ethereum/solidity/blob/3ed9a38abc9c9babf50a25f56bff2411da9cdfe7/libsolidity/analysis/TypeChecker.cpp#L2233

**Comment by matheusaaguiar** [comment]
This seems to be the same problem of #13101 and it should be fixed by #13299.

**Comment by cameel** [comment]
#13299 fixed it. Closing.
